clients/src/main/java/org/apache/kafka/common/network/SaslChannelBuilder.java

```diff
@@ -57,8 +55,7 @@ public void configure(Map<String, ?> configs) throws KafkaException {
         try {
             this.configs = configs;
             this.loginManager = LoginManager.acquireLoginManager(loginType, configs);
-            this.principalBuilder = (PrincipalBuilder) Utils.newInstance((Class<?>) configs.get(SslConfigs.PRINCIPAL_BUILDER_CLASS_CONFIG));
-            this.principalBuilder.configure(configs);
+            this.principalBuilder = ChannelBuilders.createPrincipalBuilder(configs);
```

Comment:
It seems that we don't need to maintain principalBuilder in SaslChannelBuilder since both SaslServerAuthenticator and SaslClientAuthenticator ignore principalBuilder in configure().